### PR TITLE
Knock feature: add UI to knock on spaces with the join rule 'knock'

### PR DESCRIPTION
**Pull request by aofn**
- [x] check if spaces have the joine rule `knock`
- [x] add component to let users request access to the space
- [x] add component to answer knocks #114
- [x] remember and display knocking status (optional)

A door icon is displayed in the explore header, if a space has the join rule `knock` or `knock_restricted`. Not yet the most obvious icon for asking permission for a room, so very open to suggestions.

### CODE REVIEW

File: components/UI/CopyToClipboard.js [hunk 1]
```diff
@@ -12,8 +13,7 @@ const CopyToClipboard = ({ content, title }) => {
     const copyToClipboard = async () => {
         navigator.clipboard.writeText(content);
         setWasContentCopied(true);
-        await new Promise(r => setTimeout(r, 2000));
-        setWasContentCopied(false);
+        await waitFor(() => setWasContentCopied(false));
```
Comment (fnwbr) [inline components/UI/CopyToClipboard.js:16]: Actually, when I look at all the use cases of `waitFor()` maybe I do want to get back to my comment about `_.delay` -- why do we actually `await` here in this case?

What I'm saying is: for all use cases here I think we can just use `_.delay`. I'll give two examples:

**First example**

This here would just be:
```js
_.delay(() => setWasContentCopied(false), 2500);
```

No need for `await` as far as I can tell.

Which also means we can remove the `async` from `const copyToClipboard = async () => {`.

File: components/UI/InviteUsersToMatrixRoom/index.js [hunk 2]
```diff
@@ -102,10 +103,10 @@ export const InviteUserToMatrixRoom = ({ roomId, onSuccess }) => {
 
         // if everything is okay, we let the user know and exit the view.
         successAmount > 0 && setUserFeedback(<Trans t={t} i18nKey="invitedUser" count={successAmount}>{ { successAmount } } user was invited and needs to accept your invitation</Trans>);
-        await new Promise(() => setTimeout(() => {
+        await waitFor(() => {
```
Comment (fnwbr) [inline components/UI/InviteUsersToMatrixRoom/index.js:106]: **Second example**

This here would just be:
```js
_.delay(() => {
    clearInputs();
    if (onSuccess && successAmount === selectedUsers.length) onSuccess();
}, 3000);
```

Comment (fnwbr) [reply components/UI/InviteUsersToMatrixRoom/index.js:106]: ... and so on.
I think the same thing works for all instances of `waitFor()` below. Correct me if I am wrong though.

**Review by fnwbr — CHANGES_REQUESTED** [review]

File: components/UI/InviteUsersToMatrixRoom/index.js (hunk 2 above)
Comment (aofn) [reply components/UI/InviteUsersToMatrixRoom/index.js:106]: yeah looks like it does exactly the same, i'll change them

**Review by fnwbr — CHANGES_REQUESTED** [review]
image

1. It does not make sense to ever show "Invite" and "Knock" at the same time. If it's a room we can knock on, then we cannot invite users. If it's a room we can invite users to, then there's no point in knocking.
2. The "knock" door also shows up for rooms that we are a part of already.

**Pull request merged**